File: mainActivity/src/main/java/com/sierrawireless/avphone/task/GetUserResult.java
package com.sierrawireless.avphone.task;

import net.airvantage.model.AvError;
import net.airvantage.model.User;

public class GetUserResult {

    private AvError error;

    private User user;

    GetUserResult(AvError error) {
        this.error = error;
    }

    GetUserResult(final User user) {
        this.user = user;
    }

    public boolean isError() {
        return error != null;
    }

    public User getUser() {
        return user;
    }

    public AvError getError() {
        return error;
    }
}

File: mainActivity/src/main/java/com/sierrawireless/avphone/task/GetUserResult.kt
package com.sierrawireless.avphone.task

import net.airvantage.model.AvError
import net.airvantage.model.User

class GetUserResult {

    var error: AvError? = null

    var user: User? = null

    val isError: Boolean
        get() = error != null

    constructor(error: AvError) {
        this.error = error
    }

    constructor(user: User) {
        this.user = user
    }
}
